apps/atqr/api/src/app/challenge.controller.ts

```diff
@@ -146,9 +153,40 @@ export class ChallengeController {
   }
 
   // TODO Implement change supervisor endpoint and fix return
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  private changeSupervisor(id: string): Challenge {
-    return {} as Challenge;
+  private async updateSupervisorStatus(
+    id: Guid,
+    updateSupervisorDto: UpdateSupervisorStatusDto
+  ): Promise<void> {
+    const challenge = await this.challengeRepository.findUnique(id);
+    challenge.updateSupervisorStatus(updateSupervisorDto.SupervisorEnum);
+    switch (true) {
+      case updateSupervisorDto.SupervisorEnum == SupervisorEnum.notInvited: {
+        const email = new SupConfirmation(challenge.player);
+        this.emailService.sendMail(email);
+        break;
+      }
+      case updateSupervisorDto.SupervisorEnum ==
+        SupervisorEnum.askedIfTheGoalIsAccomplished: {
+        const email = new DeadLineEmail(challenge.player);
+        this.emailService.sendMail(email);
+        break;
+      }
+      case updateSupervisorDto.SupervisorEnum ==
+        SupervisorEnum.repliedTheGoalWasSuccess: {
+        const url = `http://localhost:3333/api/challenge/${id}`;
+        const data = { id: id, status: ChallengeStatus.Completed };
+        this.httpService.axiosRef.patch(url, data);
```

Comment:
What is the reasoning for making an external API call for ourselves? This is terribly inefficient! It would make sense in a microservices based architecture but not here.